homeassistant/components/todo/__init__.py

```diff
@@ -57,24 +57,45 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     )
     component.async_register_entity_service(
         "update_item",
-        TASK_ITEM_UPDATE_FIELDS,
+        vol.All(
+            cv.make_entity_service_schema(TASK_ITEM_UPDATE_FIELDS),
+            cv.has_at_least_one_key("uid", "summary"),
+        ),
         _async_update_todo_item,
         required_features=[TodoListEntityFeature.UPDATE_TODO_ITEM],
     )
     component.async_register_entity_service(
         "delete_item",
-        {
-            vol.Required("uids"): vol.All(cv.ensure_list, [cv.string]),
-        },
+        vol.All(
+            cv.make_entity_service_schema(
+                {
+                    vol.Optional("uid"): vol.All(cv.ensure_list, [cv.string]),
+                    vol.Optional("summary"): vol.All(cv.ensure_list, [cv.string]),
+                }
+            ),
+            cv.has_at_least_one_key("uid", "summary"),
+        ),
         _async_delete_todo_items,
         required_features=[TodoListEntityFeature.DELETE_TODO_ITEM],
     )
     component.async_register_entity_service(
         "move_item",
-        {
-            vol.Required("uid"): cv.string,
-            vol.Optional("previous"): cv.string,
-        },
+        vol.All(
+            cv.make_entity_service_schema(
+                {
+                    vol.Optional("uid"): cv.string,
+                    vol.Optional("summary"): cv.string,
+                    vol.Optional("previous_uid"): cv.string,
+                    vol.Optional("previous_summary"): cv.string,
+                }
+            ),
+            cv.has_at_least_one_key("uid", "summary"),
+            cv.has_at_most_one_key("uid", "summary"),
+            cv.has_at_most_one_key("previous_uid", "previous_summary"),
+            # Don't mix and match uid and summary across current and previous
+            cv.has_at_most_one_key("uid", "previous_summary"),
+            cv.has_at_most_one_key("summary", "previous_uid"),
```

Comment:
It's easier to read if we always start with the same pair type.
```suggestion
            cv.has_at_most_one_key("previous_uid", "summary"),
```